### DISCUSSION TITLE
Текущий механизм инициализации WidgetModel усложняет написание виджет-тестов

### DISCUSSION THREAD

**Discussion by saturov** [surfstudio/SurfGear#167] [General]
На данный момент действует конвенция, согласно которой принято создавать WidgetModel непосредственно при инициализации класса виджета. В супер-конструктор WM передается так называемый WidgetModelBuilder, который производит экземпляр WidgetModel. На этом этапе осуществляется поставка в WidgetModel необходимые зависимости.

Такой механизм служит препятствием для тестирования виджета в изоляции, так как для этого требуется подменять зависимости WidgetModel на моковые. Этой возможности при текущем подходе мы лишены.

Очевидным, но некрасивым решением, является вынос создания WidgetModel из виджета на уровень описания дерева виджетов. Минус в том, что мы получим смешение DI и UI. 

Предлагаю подумать, существуют ли более приемлемые решения.

```dart
class AddEditScreen extends CoreMwwmWidget {
  AddEditScreen({
    Key? key,
    TodoEntity? todoEntity,
  }) : super(
            key: key,
            widgetModelBuilder: (context) => AddEditScreenWM(
                  Navigation(context),
                  context.read<AppProvider>().todosRepository,
                  todoEntity,
                ));
```

**Comment by Luanari** [comment]
Безусловно умный, но тестовый комментарий для демонстрации оповещений в Slack.
